### DISCUSSION TITLE
Shared storage usage increasing without..usage

### DISCUSSION THREAD

**Discussion by devsleeper** [community/community#77039] [Actions] [answered]
### Select Topic Area

Bug

### Body

I’ve started to receive notification via email regarding my supposed usage of a service I have not been using in a week.  It started notifying at 75% then 90, now 100, and I still haven’t made any commits like artifact uploads or lfs related changes in general.  Very concerning!

I downloaded a usage report and the most recent reports were about a legacy (NOT being used) repo.

What can I do to recover that which has been used without my knowledge or consent?  I’m still a month in to my assessment and can’t reasonably consider spending money on services I can’t trust will bill me accurately..  (I also think the layout of the reports could be better as well maybe it was how I opened it but plain text feels very dirty and error prone.  Why not formatted in a more readable structure?

Productivity has essentially been halted until I can get some answers which is frustrating.

thanks for taking the time.

Edit:  Looks like I'm not the only one seeing this snowball:
https://github.com/actions/upload-artifact/issues/161#issuecomment-1809130576

**Comment by corhere** [comment, marked as answer]
I ran into this myself. After some back and forth with GitHub Support, it turned out to be because I had some Actions workflow runs with artifacts that had not expired. What I've learned:
1. Changes to the artifact retention period only applies to _new_ workflow runs. The expiry date of existing runs does not change.
2. Disabling Actions on a repository only hides the Actions tab from the Web UI. The workflow runs are hidden, but the storage usage continues to accumulate.
3. There is no way to bulk-delete workflow runs or artifacts. Deleting runs can be scripted, but can hit rate limits. I resorted to the following script, and will find out soon whether it made the usage stop.
  ```gh run list --json databaseId --jq '.[].databaseId' | xargs -n 1 gh run delete```

It is very easy to start getting billed for usage, but really really hard to make it stop. In other words, GitHub Actions is full of dark patterns.

**Reply by devsleeper** [reply]
Thanks, I realize now while I had not done anything recently I did have artifacts sitting idle from previous runs.  

Please confirm or correct me if I'm wrong;  does storage/cost increase on its own over time then?  how is that even possible?  Is 500 mb not 500 mb,  and I should be billed accordingly?   I get this is probably a server somewhere that's running constantly.  

I have a server with dedicated storage I pay a flat rate for,  azure functions with a set storage cap/per run cost and a clear (live) graph of any increase in usage or cost.. what is so special about GitHub they can accumulate like this?  It feels very dark indeed.

It would make sense if I was being billed to any size increase.   The fact I received a notification stating I used 100% of my storage a Week after any activity when I was at ~70% – how can that be justified?  How do you manage that? It's such a sudden and unexpected spike in growth.

I will try the script mentioned I appreciate it. In the future I plan to automatically delete artifacts after they've been used (silly me) in the hopes it mitigates the snowball effect that seems to take place with storage.

**Reply by corhere** [reply]
Storage cost does accumulate over time. [It is billed in units of GB-hours.](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#sample-storage-cost-calculation) Storing 500 MB for thirty days costs the same as storing 15,000 MB for one day. You are paying to rent disk space that could be used by someone else after you are done with it.

**Reply by KotM** [reply]
@devsleeper thank you for the script! I improved it a bit by increasing returned workflows limit (it was 20 by default) and skipping current workflow (just to avoid warning message):

`gh run list --json databaseId --jq '.[].databaseId' --limit 500 | tail -n +2 | xargs -n 1 gh run delete`

P.S. How ridiculous this whole calculation is... :(

**Reply by relativityboy** [reply]
How does one purge the artifacts with longer expiration periods?

**Reply by KotM** [reply]
> How does one purge the artifacts with longer expiration periods?

the script above purges all artifacts, no matter expiration period.

**Reply by faisalsheiko986-png** [reply]
هاي

**Reply by kh0861958-hash** [reply]
Please my Easypasa account on YouTube tv my house so destroy please help me 03185223609 Easypasa Uzmabibi

**Reply by Aglahir** [reply]
PowerShell version:

`gh run list --json databaseId --jq '.[].databaseId' | ForEach-Object { gh run delete $_ }`

You can run in parallel if there are many:
`gh run list --json databaseId --jq '.[].databaseId' |  ForEach-Object -Parallel { gh run delete $_ } -ThrottleLimit 5`

**Comment by Ppeepost4489** [comment]
https://github.com/orgs/community/discussions/74772?sort=new#discussion-5830798

**Comment by Ppeepost4489** [comment]
> ### Select Topic Area
> 
> Bug
> 
> ### Body
> 
> I’ve started to receive notification via email regarding my supposed usage of a service I have not been using in a week.  It started notifying at 75% then 90, now 100, and I still haven’t made any commits like artifact uploads or lfs related changes in general.  Very concerning!
> 
> I downloaded a usage report and the most recent reports were about a legacy (NOT being used) repo.
> 
> What can I do to recover that which has been used without my knowledge or consent?  I’m still a month in to my assessment and can’t reasonably consider spending money on services I can’t trust will bill me accurately..  (I also think the layout of the reports could be better as well maybe it was how I opened it but plain text feels very dirty and error prone.  Why not formatted in a more readable structure?
> 
> Productivity has essentially been halted until I can get some answers which is frustrating.
> 
> thanks for taking the time.
> 
> Edit:  Looks like I'm not the only one seeing this snowball:
> https://github.com/actions/upload-artifact/issues/161#issuecomment-1809130576

**Comment by Ppeepost4489** [comment]
> ### Select Topic Area
> 
> Bug
> 
> ### Body
> 
> I’ve started to receive notification via email regarding my supposed usage of a service I have not been using in a week.  It started notifying at 75% then 90, now 100, and I still haven’t made any commits like artifact uploads or lfs related changes in general.  Very concerning!
> 
> I downloaded a usage report and the most recent reports were about a legacy (NOT being used) repo.
> 
> What can I do to recover that which has been used without my knowledge or consent?  I’m still a month in to my assessment and can’t reasonably consider spending money on services I can’t trust will bill me accurately..  (I also think the layout of the reports could be better as well maybe it was how I opened it but plain text feels very dirty and error prone.  Why not formatted in a more readable structure?
> 
> Productivity has essentially been halted until I can get some answers which is frustrating.
> 
> thanks for taking the time.
> 
> Edit:  Looks like I'm not the only one seeing this snowball:
> https://github.com/actions/upload-artifact/issues/161#issuecomment-1809130576

**Comment by shams548** [comment]
Hi update future

**Comment by 3limin4t0r** [comment]
Though the accepted answer clears the artifacts, you'll run into the same issue again if you do not update the retention period of your GitHub action. The artifact retention period is 90 days by default, to update this navigate to your `.github/workflows/*.yml` and find the `actions/upload-artifact@v4` section. Then create or lower the `retention-days`. (See [Retention Period](https://github.com/actions/upload-artifact?tab=readme-ov-file#retention-period))

> ### Retention Period
> 
> Artifacts are retained for 90 days by default. You can specify a shorter retention period using the `retention-days` input:
> 
> ```yaml
>   - name: Create a file
>     run: echo "I won't live long" > my_file.txt
> 
>   - name: Upload Artifact
>     uses: actions/upload-artifact@v4
>     with:
>       name: my-artifact
>       path: my_file.txt
>       retention-days: 5
> ```
> 
> The retention period must be between 1 and 90 inclusive. For more information see [artifact and log retention policies](https://docs.github.com/en/free-pro-team@latest/actions/reference/usage-limits-billing-and-administration#artifact-and-log-retention-policy).

**Comment by MVPuknowme** [comment]
https://sandbox.plaid.com (Sandbox)
https://production.plaid.com (Production)

**Comment by vipgmboss** [comment]
https://github.com/orgs/community/discussions/77039

**Comment by bossbdpro1** [comment]
Oops! An Error Occurred
The server returned a "500 Internal Server Error".
Something is broken. Please let us know what you were doing when this error occurred. We will fix it as soon as possible. Sorry for any inconvenience caused.

**Comment by redross15th-hash** [comment]
> ### Select Topic Area
> Bug
> 
> ### Body
> I’ve started to receive notification via email regarding my supposed usage of a service I have not been using in a week. It started notifying at 75% then 90, now 100, and I still haven’t made any commits like artifact uploads or lfs related changes in general. Very concerning!
> 
> I downloaded a usage report and the most recent reports were about a legacy (NOT being used) repo.
> 
> What can I do to recover that which has been used without my knowledge or consent? I’m still a month in to my assessment and can’t reasonably consider spending money on services I can’t trust will bill me accurately.. (I also think the layout of the reports could be better as well maybe it was how I opened it but plain text feels very dirty and error prone. Why not formatted in a more readable structure?
> 
> Productivity has essentially been halted until I can get some answers which is frustrating.
> 
> thanks for taking the time.
> 
> Edit: Looks like I'm not the only one seeing this snowball: [actions/upload-artifact#161 (comment)](https://github.com/actions/upload-artifact/issues/161#issuecomment-1809130576)

**Reply by Git-for-mo** [reply]
I paid for a Pro subscription to help me with my work flow in LaTex. When I need a more complicated command to suit my project I click the chat button. I was working on an important account when my chat was interrupted with the warning that I've exhausted my budget.

"You've exhausted your premium model quota. Please enable additional paid premium requests, upgrade to Copilot Pro+, or wait for your allowance to renew.

Manage Budget
Click to Retry
Changes may take a few minutes to take effect."

So...Co-Pilot Pro isn't enough? I need to pay for Pro+???

**Comment by ARCH-angel93** [comment]
it can indicate the requirements of some backend work or sub-process flow, searching for any repositories on backend notifications or backend procedures or frameworks to use within this can apply more corrective means of handling what may stick or reason with your storage. it has preference or not?

**Comment by ARCH-angel93** [comment]
The SHARED may be using it as a dual deposit format which can include or mimic you to alleviate the functional tolerance of processing or procedures, it is important to clarify in a written document on your device such as a workspace or on a main form of indicating where a bot handler may read this in a public or non-public way.

**Comment by ARCH-angel93** [comment]
Social structures can do this for you as well like in this comment format I may include myself as having a mutual prior inclusion to having gone through the same experience more or less multiple times in a from start as scratch-based method related to computers or development.

**Comment by SUPRANODE00** [comment]
How do I restore and regain access to my GitHub services.  I have a flagged account and I do not know why.  please email m e at DEMIEN.capsulecraft@gmail.com

**Comment by honyocokrokusuma-RLTM** [comment]
> ### Select Topic Area
> Bug
> 
> ### Body
> I’ve started to receive notification via email regarding my supposed usage of a service I have not been using in a week. It started notifying at 75% then 90, now 100, and I still haven’t made any commits like artifact uploads or lfs related changes in general. Very concerning!
> 
> I downloaded a usage report and the most recent reports were about a legacy (NOT being used) repo.
> 
> What can I do to recover that which has been used without my knowledge or consent? I’m still a month in to my assessment and can’t reasonably consider spending money on services I can’t trust will bill me accurately.. (I also think the layout of the reports could be better as well maybe it was how I opened it but plain text feels very dirty and error prone. Why not formatted in a more readable structure?
> 
> Productivity has essentially been halted until I can get some answers which is frustrating.
> 
> thanks for taking the time.
> 
> Edit: Looks like I'm not the only one seeing this snowball: [actions/upload-artifact#161 (comment)](https://github.com/actions/upload-artifact/issues/162#issuecomment-1809130576)

**Comment by kartik99917** [comment]
I’m experiencing the exact same issue, and it’s becoming very frustrating.

My shared storage usage keeps increasing even though I haven’t been actively using GitHub Actions. I haven’t pushed any workflows recently, I haven’t uploaded any new artifacts, and I even checked my repositories for old workflow runs that might still be consuming storage.

Here’s what I’ve already tried:

* Deleted old workflow artifacts.
* Checked all repositories, including archived and inactive ones.
* Verified that no workflows are currently running.
* Reduced artifact retention periods where possible.
* Removed unnecessary caches and old workflow logs.
* Reviewed the billing and storage usage pages multiple times.
* Waited for the storage metrics to refresh, but the reported usage still doesn’t decrease as expected.

Despite all of this, my shared storage either remains the same or continues to increase. It feels like some storage is being counted without any clear indication of where it’s coming from.

It would really help if GitHub provided a more detailed storage breakdown showing exactly which repository, workflow run, artifact, or cache is consuming space. Right now it’s very difficult to identify the source of the usage.

Has anyone else found a reliable solution, or is this still an unresolved issue on GitHub’s side? Any guidance from the GitHub team would be greatly appreciated.

**Comment by KotM** [comment]
The limit will be recalculated only at the beginning of next month.

**Comment by wcanchanMG** [comment]
El límite no se recalculará aunque elimine los artefactos y he esperado 2 semanas y nada, estoy bloqueado, además tengo cuenta GitHub Pro.

**Reply by MVPuknowme** [reply]
Gracias 😊
